### PR TITLE
[CLEANUP]: Version 2 Preparation

### PR DESCRIPTION
**Pull request by amadolid**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
-VERSION = "0.0.1"
+VERSION = "2.0.0"
 
 setup(
     name="jaclang-jaseci",
     version=VERSION,
     packages=find_packages(include=["jaclang_jaseci", "jaclang_jaseci.*"]),
     install_requires=[
-        "jaclang",
+        "jaclang==0.7.5",
```
Comment (ypkang) [inline setup.py]: We should figure out an elegant way to handle version comparabilities between jaclang and jaclang-jaseci. Maybe pluggy has some builtin way of handling it.

File: setup.py [hunk 2]
```diff
@@ -21,19 +21,22 @@
...
     ],
     package_data={},
     entry_points={
         "jac": [],
     },
-    author="Jason Mars, Yiping Kang, Alexie Madolid",
+    author="Jason Mars",
     author_email="jason@jaseci.org",
-    url="https://github.com/Jaseci-Labs/jaclang-jaseci",
+    url="https://github.com/Jaseci-Labs/jaclang",
```
Comment (ypkang) [inline setup.py:42]: Update this URL to jaclang-jaseci

File: setup.py [hunk 3]
```diff
@@ -5,14 +5,14 @@
 from setuptools import find_packages, setup
 
 
-VERSION = "0.0.1"
+VERSION = "2.0.0"
```
Comment (ypkang) [inline setup.py:8]: I think we should start with 0.1. Make it consistent with jaclang-jaseci. Since we haven't pushed to pypi yet, we can start low.

**Review by ypkang — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (ChrisIsKing) [reply setup.py]: Can we also relax a few of these dependencies? Are the specific versions required to support jaseci?

File: setup.py (hunk 3 above)
Comment (ypkang) [reply setup.py:8]: I added a pyproject.toml file to use peotry for building this. So we should add the dependencies there and remove the setup.py

File: setup.py (hunk 1 above)
Comment (ypkang) [reply setup.py]: yeah thats a good idea.

File: setup.py [hunk 4]
```diff
@@ -5,35 +5,39 @@
...
+        "types-passlib==1.7.7.20240327",
+        "email-validator==2.2.0",
+        "orjson==3.10.6",
+        "redis==5.0.7",
+        "types-redis==4.6.0.20240425",
         "python-multipart==0.0.9",
         "httpx==0.27.0",
+        "sendgrid==6.11.0",
+        "nest-asyncio==1.6.0",
+        "fastapi-sso==0.15.0",
```
Comment (ypkang) [inline setup.py:34]: I think we can relax some of these. Generally we can do any minor version that is more recent than the version listed here.

So for example, we are currently pinning pydantic at 2.8.2. I think we can relax it to be `pydantic>=2.8.2, <2.9`

**Review by ypkang — CHANGES_REQUESTED** [review]

**Pull request closed without merging**